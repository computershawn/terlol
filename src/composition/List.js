import React from 'react';
import Card from './Card';
import './List.css';


class List extends React.Component {
  static defaultProps = {
    header: 'no header',
    cards: [],
  };
  render() {
    //console.log('Using a class component!')
    const listOfCards = this.props.cards.map(function(card, index) {
        return <Card key={index} title={card.title} content={card.content}/>
    });
    return (
        <section className="List">
            <header className="List-header">
                <h2>{this.props.header}</h2>
            </header>
            <div className="List-cards">
                {listOfCards}
            </div>
        </section>
    )
  }
}

export default List;






