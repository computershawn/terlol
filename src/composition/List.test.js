import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('List component', () => {
    const cards = [
        { title: 'First card', content: 'lorem ipsum' }, 
        { title: 'Second card', content: 'lorem ipsum' },
        { title: 'Fifth card', content: 'lorem ipsum' }
    ];

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List key="1" header="First list" cards={cards} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<List key="1" header="First list" cards={cards} />)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
});