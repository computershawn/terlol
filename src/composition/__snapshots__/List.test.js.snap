exports[`List component renders the UI as expected 1`] = `
<section
  className="List">
  <header
    className="List-header">
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards">
    <div
      className="Card">
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card">
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card">
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
